### PR TITLE
XWiki 10.4 has been released

### PR DESCRIPTION
**Pull request by vmassol**

### CODE REVIEW

**Comment by tianon** [comment]
<details>
<summary>Diff:</summary>

```diff
diff --git a/_bashbrew-list b/_bashbrew-list
index f86906e..244c93f 100644
--- a/_bashbrew-list
+++ b/_bashbrew-list
@@ -8,9 +8,9 @@ xwiki:9.11.3-postgres-tomcat
 xwiki:10
 xwiki:10-mysql-tomcat
 xwiki:10-postgres-tomcat
-xwiki:10.3
-xwiki:10.3-mysql-tomcat
-xwiki:10.3-postgres-tomcat
+xwiki:10.4
+xwiki:10.4-mysql-tomcat
+xwiki:10.4-postgres-tomcat
 xwiki:latest
 xwiki:lts
 xwiki:lts-mysql
diff --git a/xwiki_latest/Dockerfile b/xwiki_latest/Dockerfile
index 5a3d2ba..86dc5f7 100644
--- a/xwiki_latest/Dockerfile
+++ b/xwiki_latest/Dockerfile
@@ -44,9 +44,9 @@ RUN apt-get update && \
 # Install XWiki as the ROOT webapp context in Tomcat
 # Create the Tomcat temporary directory
 # Configure the XWiki permanent directory
-ENV XWIKI_VERSION=10.3
+ENV XWIKI_VERSION=10.4
 ENV XWIKI_URL_PREFIX "http://maven.xwiki.org/releases/org/xwiki/platform/xwiki-platform-distribution-war/${XWIKI_VERSION}"
-ENV XWIKI_DOWNLOAD_SHA256 46e76e6d74fc59c00d09bbe03da36b8ff223a9cfa6e84cf2eff3001922ac715f
+ENV XWIKI_DOWNLOAD_SHA256 4e9575813cb8e3c5c6fcb76f7a2df8bb6d0ec36b188235b37612184cb7b56a1a
 RUN rm -rf /usr/local/tomcat/webapps/* && \
   mkdir -p /usr/local/tomcat/temp && \
   mkdir -p /usr/local/xwiki/data && \
diff --git a/xwiki_latest/xwiki/docker-entrypoint.sh b/xwiki_latest/xwiki/docker-entrypoint.sh
index 4e7f232..e72e445 100755
--- a/xwiki_latest/xwiki/docker-entrypoint.sh
+++ b/xwiki_latest/xwiki/docker-entrypoint.sh
@@ -121,12 +121,6 @@ function configure() {
   safesed "replacecontainer" $DB_HOST /usr/local/tomcat/webapps/ROOT/WEB-INF/hibernate.cfg.xml
   safesed "replacedatabase" $DB_DATABASE /usr/local/tomcat/webapps/ROOT/WEB-INF/hibernate.cfg.xml
 
-  echo '  Using filesystem-based attachments...'
-  xwiki_set_cfg 'xwiki.store.attachment.hint' 'file'
-  xwiki_set_cfg 'xwiki.store.attachment.versioning.hint' 'file'
-  xwiki_set_cfg 'xwiki.store.attachment.recyclebin.content.hint' 'file'
-  echo '  Using filesystem-based deleted documents...'
-  xwiki_set_cfg 'xwiki.store.recyclebin.content.hint' 'file'
   echo '  Generating authentication validation and encryption keys...'
   xwiki_set_cfg 'xwiki.authentication.validationKey' "$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1)"
   xwiki_set_cfg 'xwiki.authentication.encryptionKey' "$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1)"
````

</details>

**Comment by tianon** [comment]
LGTM

Build test of #4373; efd55f0f; `amd64` (`xwiki`):

```console

$ bashbrew build xwiki:9
Building bashbrew/cache:afb0ee66cdb771d10fe98490d1535b53281056c702080bcd4abd5a0c9b38a9ca (xwiki:9)
Tagging xwiki:9
Tagging xwiki:9.11
Tagging xwiki:9.11.3
Tagging xwiki:9-mysql-tomcat
Tagging xwiki:mysql-tomcat
Tagging xwiki:lts-mysql-tomcat
Tagging xwiki:lts-mysql
Tagging xwiki:lts

$ test/run.sh xwiki:9
testing xwiki:9
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...passed
	'override-cmd' [4/4]...passed


$ bashbrew build xwiki:9-postgres-tomcat
Building bashbrew/cache:aaf7452028365a0dbff562c1f6ecaaa3bbe8528d23f0a5581fa61c8b625c201c (xwiki:9-postgres-tomcat)
Tagging xwiki:9-postgres-tomcat
Tagging xwiki:9.11-postgres-tomcat
Tagging xwiki:9.11.3-postgres-tomcat
Tagging xwiki:postgres-tomcat
Tagging xwiki:lts-postgres-tomcat
Tagging xwiki:lts-postgres

$ test/run.sh xwiki:9-postgres-tomcat
testing xwiki:9-postgres-tomcat
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...passed
	'override-cmd' [4/4]...passed


$ bashbrew build xwiki:10
Building bashbrew/cache:9dbe9b8b7ac90efd7a1a4246ed23dfb4a69c68451f8305880c2964f9261b11a4 (xwiki:10)
Tagging xwiki:10
Tagging xwiki:10.4
Tagging xwiki:10-mysql-tomcat
Tagging xwiki:10.4-mysql-tomcat
Tagging xwiki:stable-mysql-tomcat
Tagging xwiki:stable-mysql
Tagging xwiki:stable
Tagging xwiki:latest

$ test/run.sh xwiki:10
testing xwiki:10
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...passed
	'override-cmd' [4/4]...passed


$ bashbrew build xwiki:10-postgres-tomcat
Building bashbrew/cache:5e9d97416408336ff93f6a5a9a22ffbadbecc373086e3fc145a533a486deca7c (xwiki:10-postgres-tomcat)
Tagging xwiki:10-postgres-tomcat
Tagging xwiki:10.4-postgres-tomcat
Tagging xwiki:stable-postgres-tomcat
Tagging xwiki:stable-postgres

$ test/run.sh xwiki:10-postgres-tomcat
testing xwiki:10-postgres-tomcat
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...passed
	'override-cmd' [4/4]...passed

```

**Comment by vmassol** [comment]
Thanks Tianon

**Comment by TheCodingLand** [comment]
It seems 10-postgres-tomcat version is still pointing to a branch with 10.3 on docker hub for now.

File: library/xwiki [hunk 1]
```diff
@@ -9,10 +9,10 @@ Tags: 9-postgres-tomcat, 9.11-postgres-tomcat, 9.11.3-postgres-tomcat, postgres-
...
 Directory: 9/postgres-tomcat
 
-Tags: 10, 10.3, 10-mysql-tomcat, 10.3-mysql-tomcat, stable-mysql-tomcat, stable-mysql, stable, latest
-GitCommit: 1b02c423a736424a30c390ddb77a0128aa925f17
+Tags: 10, 10.4, 10-mysql-tomcat, 10.4-mysql-tomcat, stable-mysql-tomcat, stable-mysql, stable, latest
+GitCommit: 0fe4bba54861f755d6a188c401b470e303bd4b34
 Directory: 10/mysql-tomcat
 
-Tags: 10-postgres-tomcat, 10.3-postgres-tomcat, stable-postgres-tomcat, stable-postgres
+Tags: 10-postgres-tomcat, 10.4-postgres-tomcat, stable-postgres-tomcat, stable-postgres
```
Comment (vmassol) [inline library/xwiki]: Bug here! I'll send another PR to fix it.

**Comment by vmassol** [comment]
@TheCodingLand  Thanks a lot for reporting the bug, I've sent a new PR to fix it at https://github.com/docker-library/official-images/pull/4442

**Pull request merged**